### PR TITLE
add prefix and suffix to input

### PR DESCRIPTION
**Pull request by fieldsco**

### CODE REVIEW

**Comment by fieldsco** [comment]
[(2) Add suffix and prefix options to Input component](https://trello.com/c/3HkjoHWe/554-2-add-suffix-and-prefix-options-to-input-component)

**Comment by forgingdestiny** [comment]
[(0) Exchange Rates Task KeyError](https://trello.com/c/TbAOIyXa/1088-0-exchange-rates-task-keyerror)

File: src/components/Input/Input.js [hunk 1]
```diff
@@ -83,6 +95,20 @@ class Input extends React.Component {
     onKeyDown: PropTypes.func,
   }
 
+  componentDidMount = () => {
+    let inputStyles = {}
+    if (this.props.prefix) {
+      // Add 24 to accomodate for left and right padding of prefix (16+8)
```
Comment (samageloff) [inline src/components/Input/Input.js]: Spelling

File: src/components/Input/Input.js [hunk 2]
```diff
@@ -83,6 +95,20 @@ class Input extends React.Component {
     onKeyDown: PropTypes.func,
   }
 
+  componentDidMount = () => {
+    let inputStyles = {}
```
Comment (samageloff) [inline src/components/Input/Input.js]: Drop a line break please

File: src/components/Input/Input.js [hunk 3]
```diff
@@ -91,53 +118,40 @@ class Input extends React.Component {
...
-    var inputClass = cx(style['input-component'], this.props.optClass, disabledClass)
+    const disabledClass = this.props.disabled ? style['input-disabled'] : null
+    const prefixClass = prefix ? style['prefix'] : null
+    const suffixClass = suffix ? style['suffix'] : null
+    const inputClass = cx(style['input-component'], optClass, disabledClass, prefixClass, suffixClass)
 
     return (
       <div className={inputClass}>
-        { label ? <label>{label}</label> : null }
+        {label && <label>{label}</label>}
```
Comment (samageloff) [inline src/components/Input/Input.js]: This is sick.

File: src/components/Input/Input.js [hunk 4]
```diff
@@ -91,53 +118,40 @@ class Input extends React.Component {
...
+
+        {prefix && <div ref={(c) => this._prefix = c} className={prefixClass}>{prefix}</div>}
```
Comment (samageloff) [inline src/components/Input/Input.js]: What if there's a prefix and a suffix?

File: test/Input.js [hunk 5]
```diff
@@ -10,11 +10,19 @@ describe('Input', () => {
...
     expect(wrapper.childAt(1).props().value).to.equal('Initial value.')
   })
 
+  it('should shallow render a prefix and suffix', () => {
+    wrapper = shallow(<Input prefix='$' suffix='days' />)
+    expect(wrapper.find('div')).to.have.length(3)
+    expect(wrapper.childAt(0).props().className).to.equal('prefix')
+    expect(wrapper.childAt(2).props().className).to.equal('suffix')
+  })
+
```
Comment (samageloff) [inline test/Input.js]: We should add a test to confirm `componentDidMount`. Can you mock `gbrc`?

**Comment by samageloff** [comment]
👍

**Pull request merged**